client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -96,13 +97,10 @@ class JetpackOnboardingStatsStep extends React.Component {
 		const connectUrl = addQueryArgs(
 			{
 				from: 'jpo',
-				redirect_after_auth: [ basePath, stepName, siteSlug ].join( '/' ),
-				url: siteUrl,
-				// TODO: add a parameter to the JPC to redirect back to this step after completion
-				// and in the redirect URL include the ?action=activate_stats parameter
-				// to actually trigger the stats activation action after getting back to JPO
+				redirect_after_auth: addQueryArgs( { action: 'activate_stats' }, window.location.href ),
```

Comment:
Can we avoid `window.location.href` here? Why not use `[ basePath, stepName, siteSlug ].join( '/' )`? I thought we were using relative redirects for redirects elsewhere, won't it work here?

Reply:
Unfortunately, any domain-less redirect URLs will not be considered valid, `wp_validate_redirect` will not allow them and will fall back to the default redirect URL:

https://github.com/Automattic/jetpack/blob/a4f4f2d4f3827ad380061aed03d291fc1625bf21/class.jetpack.php#L4480

Reply:
Hmm, there's `config( 'hostname' )` and `config( 'port' )`. Maybe we need a helper function to use those. (But that's a detail at this point.)

Reply:
Do you mean as additional parameters?

Reply:
Well, to assemble the redirect URL from known bits (instead of `window.location.href`). I'd hardwire the `config` bits into a helper function -- e.g. `makeAbsolureUrl( [ basePath, stepName, siteSlug ].join( '/' ) )`.

Reply:
Ah - sounds interesting - feel free to push in the PR, it's yours after all 😉